src/game.cpp

```diff
@@ -848,6 +851,59 @@ bool game::start_game()
     return true;
 }
 
+void game::place_starting_vehicle( const vproto_id &starting_vehicle )
+{
+    vehicle veh( starting_vehicle );
+    std::vector<std::string> omt_search_types;
+    if( veh.max_ground_velocity() > 0 ) {
+        omt_search_types.push_back( "road" );
+        omt_search_types.push_back( "field" );
+    } else if( veh.can_float() ) {
+        omt_search_types.push_back( "river" );
+        omt_search_types.push_back( "lake" );
+    }
+    place_vehicle_nearby( starting_vehicle, u.global_omt_location().xy(), 1, 30, omt_search_types );
+
+}
+
+vehicle *game::place_vehicle_nearby( const vproto_id &id, const point &origin, int min_distance,
+                                     int max_distance, const std::vector<std::string> &omt_search_types )
+{
+    for( const std::string &search_type : omt_search_types ) {
+        omt_find_params find_params;
+        find_params.must_see = false;
+        find_params.cant_see = false;
+        find_params.types.emplace_back( search_type, ot_match_type::type );
+        // find nearest road
+        find_params.min_distance = min_distance;
+        find_params.search_range = max_distance;
+        // if player spawns underground, park their car on the surface.
+        const tripoint omt_origin( origin.x, origin.y, 0 );
+        std::vector<tripoint> goals = overmap_buffer.find_all( omt_origin, find_params );
+        for( const tripoint &goal : goals ) {
+            // try place vehicle there.
+            tinymap target_map;
+            target_map.load( omt_to_sm_copy( goal ), false );
+            tripoint origin = target_map.getlocal( sm_to_ms_copy( omt_to_sm_copy( goal ) ) ) + point(
```

Comment:
```suggestion
            const tripoint origin( SEEX, SEEY );
```
The `target_map` starts at `omt_to_sm_copy( goal )` (see line directly above where it's loaded). In map tiles this would be `sm_to_ms_copy( omt_to_sm_copy( goal ) )`.

`getlocal` will just remove that offset again.

Your currernt lines is essentially `origin = sm_to_ms_copy( omt_to_sm_copy( goal ) ) + point( ... ) - sm_to_ms_copy( omt_to_sm_copy( goal ) )`.